### PR TITLE
Add a text field which converts paste events to BBC key sequences.

### LINKED ISSUE #164: Support "paste"
Suggested over email, pasting snippets might be handy

### PR DESCRIPTION
**Pull request by scarybeasts**
Fixes #164 

Supports paste via Ctrl-V, middle-mouse (on Linux at least) and context menu -> paste.

### CODE REVIEW

**Comment by richtw1** [comment]
Is there some way to prevent Ctrl+V from doing that, as it's a real BBC key sequence (change mode)? I think it's important that any program which is actually waiting for ASCII code 22 can actually receive it.

**Comment by scarybeasts** [comment]
@richtw1 Ctrl-V goes through to the BBC unless the paste text field is focused. Focus is automatically regranted to the main BBC window upon mouseover / mousedown / etc.

**Comment by richtw1** [comment]
Ah I understand. Awesome!

**Comment by scarybeasts** [comment]
Also now fixes #223

File: main.js [hunk 1]
```diff
@@ -645,26 +653,8 @@ require(['jquery', 'underscore', 'utils', 'video', 'soundchip', 'ddnoise', 'debu
...
-                BBC.C,
-                BBC.H,
-                BBC.PERIOD,
-                BBC.SHIFT,
-                BBC.K2,
-                BBC.K2,
-                BBC.SHIFT,
-                BBC.RETURN
-            );
+            var bbcKeys = utils.stringToBBCKeys('*TAPE\nCH.""\n');
```
Comment (mattgodbolt) [inline main.js:656]: OH my word, so much better! thank you

**Pull request merged**